vm/src/builtins/type.rs

```diff
@@ -32,6 +34,28 @@ pub struct PyType {
     pub slots: PyTypeSlots,
 }
 
+#[repr(C)]
+pub struct PyHeapType {
+    pub typ: PyType,
+    pub number_methods: PyRwLock<PyNumberMethods>,
+}
+
+impl From<PyHeapType> for PyType {
+    fn from(heaptype: PyHeapType) -> Self {
+        //TODO: SAFYTY
+        unsafe { std::mem::transmute(heaptype) }
```

Comment:
This is not possible. I think this `From` impl never can exist. Their references can be casted, but not about the object body itself.